clippy_lints/src/doc.rs

```diff
@@ -258,6 +259,18 @@ fn check_word(cx: &EarlyContext, word: &str, span: Span) {
         s != "_" && !s.contains("\\_") && s.contains('_')
     }
 
+    if let Ok(url) = Url::parse(word) {
```

Comment:
I feel like we should first check for http/https or something to avoid hitting the URL parser too much